src/features/import/components/Upload/UploadFile.tsx

```diff
@@ -0,0 +1,177 @@
+import { UploadFileOutlined } from '@mui/icons-material';
+import { useDropzone } from 'react-dropzone';
+import {
+  Box,
+  Button,
+  CircularProgress,
+  IconButton,
+  Typography,
+} from '@mui/material';
+import { useCallback, useState } from 'react';
+
+import messageIds from '../../l10n/messageIds';
+import theme from 'theme';
+import useImportedFile from '../../hooks/useImportedFile';
+import { Msg, useMessages } from 'core/i18n';
+
+const UploadFile = () => {
+  const [error, setError] = useState<boolean>(false);
+  const messages = useMessages(messageIds);
+  const { parseData, loading } = useImportedFile();
+
+  const onDrop = useCallback((acceptedFiles: File[]): void => {
+    acceptedFiles.map((file: File) => {
+      parseData(file);
+    });
+  }, []);
+
+  const { getRootProps, getInputProps, open, isDragActive } = useDropzone({
+    accept: {
+      types: ['.csv', '.xls', '.xlsx'],
+    },
+    maxFiles: 1,
+    noClick: true,
+    noKeyboard: true,
+    onDrop,
+    onDropAccepted: () => setError(false),
+    onDropRejected: () => setError(true),
+    onError: () => {
+      setError(true);
+    },
+  });
+
+  return (
+    <>
+      <Box
+        {...getRootProps()}
+        p={12}
+        style={
+          loading
+            ? {
+                alignItems: 'center',
+                backgroundColor: '#1976D214',
+                borderColor: '#1976D2',
+                borderRadius: 4,
+                borderStyle: 'dashed',
+                borderWidth: 2,
+                display: 'flex',
+                flexDirection: 'column',
+                flexWrap: 'wrap',
+                margin: 16,
+                textAlign: 'center',
+              }
+            : {
+                alignItems: 'center',
```

Comment:
Many of these properties reoccur regardless of `loading` value. I would find another way to write these, either using classes with `makeStyles()` or by using ternary on the properties that do change, the same way that you have done with `error`.